lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,449 @@
+        This requires a JSON data with required user fields
+        {
+            "username": "username",
+            "password": "password",
+            "firstName": first_name,
+            "lastName": "last_name",
+            "email": "user@domain.com"
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+
+        :return: JSON Payload
+        if we succeed to add a user entry in database, the returned response_object will include the auth_token
+            response_object = {
+                    "status": "success"/"fail",
+                    "status_code": 201/status_code_int,
+                    "message": "Successfully registered."/"failure message",
+                    "auth_token": auth_token.decode(), # Will not present if failed
+                }
+        """
+        # get the post data
+        user_data = request.get_json()
+        if not user_data:
+            self.logger.warning("Invalid json object: %s", request.url)
+            abort(400, message="Invalid json object in request")
+
+        username = user_data.get("username")
+        if not username:
+            self.logger.warning("Missing username field")
+            abort(
+                400, message=f"Missing username field",
+            )
+        if username.lower() == "admin":
+            self.logger.warning("User tried to register with admin username")
+            abort(
+                400, message="Please choose another username",
+            )
+
+        password = user_data.get("password")
+        if not password:
+            self.logger.warning("Missing password field")
+            abort(
+                400, message="Missing password field",
+            )
+
+        emailID = user_data.get("email")
+        if not emailID:
+            self.logger.warning("Missing email field")
+            abort(
+                400, message="Missing email field",
+            )
+
+        firstName = user_data.get("firstName")
+        if not firstName:
+            self.logger.warning("Missing firstName field")
+            abort(
+                400, message="Missing firstName field",
+            )
+
+        lastName = user_data.get("lastName")
+        if not lastName:
+            self.logger.warning("Missing lastName field")
+            abort(
+                400, message="Missing lastName field",
+            )
+
+        # validate the email field
+        try:
+            valid = validate_email(emailID)
+            email = valid.email
+        except EmailNotValidError as e:
+            self.logger.warning("Invalid email %s", emailID)
+            abort(400, message=f"Invalid email: {emailID}")
+
+        # check if user already exist
+        user = UserModel.query.filter_by(username=user_data.get("username")).first()
+        if user:
+            abort(409, message="A user with that name already exists.")
```

Comment:
This isn't really a "conflict in the state of the object" ... it's probably more a 403 Forbidden ... "or attempting a prohibited action (e.g. creating a duplicate record where only one is allowed)"